plugins/CoreAdminHome/Commands/SetConfig.php

```diff
@@ -0,0 +1,88 @@
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Plugins\CoreAdminHome\Commands\SetConfig\ConfigSettingManipulation;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SetConfig extends ConsoleCommand
+{
+    protected function configure()
+    {
+        $this->setName('core:set-config');
+        $this->setDescription('Set one or more config settings for a particular Piwik instance.');
+        $this->addArgument('assignment', InputArgument::OPTIONAL | InputArgument::IS_ARRAY,
+            "List of config setting assignments, eg, Section.key=1 or Section.array_key[]=value");
+        $this->addOption('section', null, InputOption::VALUE_REQUIRED, 'The section the INI config setting belongs to.');
+        $this->addOption('key', null, InputOption::VALUE_REQUIRED, 'The name of the INI config setting.');
+        $this->addOption('value', null, InputOption::VALUE_REQUIRED, 'The value of the setting.');
+        $this->setHelp("This command can be used to set INI config settings on a Piwik instance.
+
+Use the --piwik-domain option to specify which instance to modify.
+
+You can set config values two ways, via --section, --key, --value or by command arguments.
+
+To use --section, --key, --value, simply supply those options. You can only set one setting this way, and you cannot
+append to arrays.
+
+To use arguments, supply one or more arguments in the following format: Section.config_setting_name=\"value\"
+'Section' is the name of the section, 'config_setting_name' the name of the setting and 'value' is the value.
+NOTE: 'value' must be JSON encoded, so Section.config_setting_name=\"value\" would work but
+Section.config_setting_name=value would not.
+
+To append to an array setting, supply an argument like this: Section.config_setting_name[]=\"value to append\"");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $section = $input->getOption('section');
+        $key = $input->getOption('key');
+        $value = $input->getOption('value');
+
+        $manipulations = $this->getAssignments($input);
+
+        $isSingleAssignment = !empty($section) && !empty($key) && !empty($value);
+        if ($isSingleAssignment) {
+            $manipulations[] = new ConfigSettingManipulation($section, $key, $value);
+        }
+
+        if (empty($manipulations)) {
+            throw new \InvalidArgumentException("Nothing to assign. Add assignments as arguments or use the "
+                . "--section, --key and --value options.");
+        }
+
+        $config = Config::getInstance();
+        foreach ($manipulations as $manipulation) {
+            $manipulation->manipulate($config);
+
+            $output->writeln("Setting [{$manipulation->getSectionName()}] {$manipulation->getName()} = {$manipulation->getValueString()}");
+        }
```

Comment:
Should we also show a warning / info that this command should be executed on every server? This would be especially useful if one runs Piwik on multiple servers. Users might forget it otherwise